frontend/src/pages/main/workspace-menu/index.tsx

```diff
@@ -6,24 +6,33 @@ import { WorkspaceMenuItem, WorkspaceMenuList } from './index.style';
 import { WorkspaceMenuProps } from './index.types';
 import DeleteModal from '../delete-modal';
 import RenameModal from '../rename-modal';
+import axios from 'axios';
 
-function WorkspaceMenu({ workspaceName }: WorkspaceMenuProps) {
-	const { isOpenModal, modalRef, toggleOpenModal, closeModal } = useModal();
+function WorkspaceMenu({ workspaceId, role, workspaceName }: WorkspaceMenuProps) {
+	const { isOpenModal, modalRef, toggleOpenModal } = useModal();
 	const [modalContent, setModalContent] = useState(<></>);
 	const openReanmeModal = () => {
+		const renameWorkspace = async (workspaceName: string) => {
+			toggleOpenModal();
+			await axios.patch(`/api/workspace/${workspaceId}/info/metadata`, { name: workspaceName });
+		};
 		toggleOpenModal();
-		setModalContent(<RenameModal closeModal={closeModal} workspaceName={workspaceName}></RenameModal>);
+		setModalContent(<RenameModal action={renameWorkspace} workspaceName={workspaceName}></RenameModal>);
 	};
 	const openDeleteModal = () => {
+		const deleteWorkspace = async () => {
+			toggleOpenModal();
+			await axios.delete(`/api/workspace/${workspaceId}`);
+		};
 		toggleOpenModal();
-		setModalContent(<DeleteModal closeModal={closeModal}></DeleteModal>);
+		setModalContent(<DeleteModal action={deleteWorkspace}></DeleteModal>);
 	};
 	return (
 		<>
 			<WorkspaceMenuList>
 				<WorkspaceMenuItem>Open</WorkspaceMenuItem>
-				<WorkspaceMenuItem onClick={openReanmeModal}>Rename</WorkspaceMenuItem>
-				<WorkspaceMenuItem onClick={openDeleteModal}>Delete</WorkspaceMenuItem>
+				{role == 2 && <WorkspaceMenuItem onClick={openReanmeModal}>Rename</WorkspaceMenuItem>}
+				{role == 2 && <WorkspaceMenuItem onClick={openDeleteModal}>Delete</WorkspaceMenuItem>}
```

Comment:
해당 부분은 같은 권한에 대해 적용되는 부분이다보니 묶어주는게 깔끔하지 않을까 하는 생각입니다!

```suggestion
				{role == 2 && 
				<>
				  <WorkspaceMenuItem onClick={openReanmeModal}>Rename</WorkspaceMenuItem>
				  <WorkspaceMenuItem onClick={openDeleteModal}>Delete</WorkspaceMenuItem>
				</>
				}
```

Reply:
아 저렇게 묶어주면 되는군요! 감사합니다